airflow/models/dagbag.py

```diff
@@ -178,7 +180,7 @@ def get_dag(self, dag_id):
                 del self.dags[dag_id]
         return self.dags.get(dag_id)
 
-    def process_file(self, filepath, only_if_updated=True, safe_mode=True):
+    def process_file(self, filepath, only_if_updated=True, safe_mode=True, check_pools=False, pools=None):
```

Comment:
Are you sure this validation must take place in DagBag? Wouldn't it be better if it was done in DagFileProcessor? Then we may not display errors for DAGs that are paused.

Reply:
I think it's better to fail fast. If my dag is invalid, I want to know about it right away, even if it's paused. We already check for import errors and dag cycles in `DagBag.process_file`, so I think it makes sense to check for missing pools here too.